src/CommonDBTM.php

```diff
@@ -667,32 +667,31 @@ public function updateInDB($updates, $oldvalues = [])
     {
         global $DB;
 
+        $tobeupdated = [];
         foreach ($updates as $field) {
             if (isset($this->fields[$field])) {
-                $DB->update(
-                    $this->getTable(),
-                    [$field => $this->fields[$field]],
-                    ['id' => $this->fields['id']]
-                );
-                if ($DB->affectedRows() == 0) {
-                    if (isset($oldvalues[$field])) {
-                        unset($oldvalues[$field]);
-                    }
-                }
-            } else {
-               // Clean oldvalues
-                if (isset($oldvalues[$field])) {
+                if (isset($oldvalues[$field]) && $this->fields[$field] == $oldvalues[$field]) {
                     unset($oldvalues[$field]);
                 }
+                $tobeupdated[$field] = $this->fields[$field];
+            } else {
+                // Clean oldvalues
+                unset($oldvalues[$field]);
             }
         }
+        $DB->update(
+            $this->getTable(),
+            $tobeupdated,
+            ['id' => $this->fields['id']]
+        );
```

Comment:
I think we should not add anything in `glpi_logs` nor reload the object if the update query fails. It probably means that an error happens and the whole query has been refused.

```suggestion
        $result = $DB->update(
            $this->getTable(),
            $tobeupdated,
            ['id' => $this->fields['id']]
        );
        if ($result === false) {
            return false;
        }
```